manager/src/main/java/io/aklivity/zilla/manager/internal/commands/install/cache/ZpmCache.java

```diff
@@ -15,238 +15,451 @@
+            Artifact artifact = new DefaultArtifact(dep.groupId, dep.artifactId, "jar", version);
+            collectRequest.addDependency(new Dependency(artifact, null));
+        }
+        repositories.forEach(collectRequest::addRepository);
+
+        DependencyResult result;
+        try
+        {
+            DependencyRequest dependencyRequest = new DependencyRequest(collectRequest, null);
+            result = repositorySystem.resolveDependencies(session, dependencyRequest);
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException("Failed to resolve dependencies", e);
+        }
+        DependencyNode root = result.getRoot();
+
+        NodeListGenerator nlg = new NodeListGenerator();
+
+        root.accept(new PreorderDependencyNodeConsumerVisitor(nlg));
+        List<DependencyNode> nodesWithDependencies = nlg.getNodesWithDependencies();
+
+        nodesWithDependencies.forEach(node ->
+        {
+            Dependency dep = node.getDependency();
+            if (dep != null)
+            {
+                final Artifact artifact = dep.getArtifact();
+                List<DependencyNode> children = node.getChildren();
+                final ZpmArtifactId id =
+                    new ZpmArtifactId(artifact.getGroupId(), artifact.getArtifactId(), artifact.getVersion());
+                final Set<ZpmArtifactId> depends = new LinkedHashSet<>();
+                children.forEach(c ->
+                {
+                    final Artifact cArtifact = c.getArtifact();
+                    final ZpmArtifactId cid =
+                        new ZpmArtifactId(cArtifact.getGroupId(), cArtifact.getArtifactId(), cArtifact.getVersion());
+                    depends.add(cid);
+                });
+                if (artifacts.stream().noneMatch(a -> a.id.equalsGroupAndArtifact(id) &&
+                    new ComparableVersion(id.version).compareTo(new ComparableVersion(a.id.version)) <= 0))
+                {
+                    artifacts.add(new ZpmArtifact(id, artifact.getFile().toPath(), depends));
+                }
+            }
+        });
 
-        return resolveDependencyArtifacts(resolvable);
+        return artifacts;
     }
 
-    private Map<ZpmDependency, String> resolveImports(
-        List<ZpmDependency> imports)
+    private RepositorySystemSession newRepositorySystemSession(
+        RepositorySystem system,
+        Path dir)
     {
-        Map<ZpmDependency, String> imported = new LinkedHashMap<>();
+        return new SessionBuilderSupplier(system)
+            .get()
+                .withLocalRepositoryBaseDirectories(dir)
+                .setRepositoryListener(new ZpmConsoleRepositoryListener())
+                .setTransferListener(new ZpmConsoleTransferListener())
+                .setConfigProperty("aether.generator.gpg.enabled", Boolean.TRUE.toString())
+                .setConfigProperty("aether.syncContext.named.factory", "noop")
```

Comment:
Are these string property names defined as constants in maven api somewhere so we could use those constants instead?